pkg/upgrade/upgrade.go

```diff
@@ -48,6 +48,10 @@ func Upgrade(cfg Config) error {
 		return fmt.Errorf("failed to parse profile installation: %w", err)
 	}
 
+	if profileInstallation.Spec.Catalog == nil {
+		return fmt.Errorf("unable to upgrade an installation that was not created from a catalog")
```

Comment:
could add a hint like "please specify spec.catalog"

Reply:
this isn't something they can fix though. You can only use the `pctl upgrade` command if you installed from a catalog entry, e.g.:
```
$ pctl add --name jake-rocks catalog/weaveworks-nginx/v0.1.0
$ pctl upgrade jake-rocks v0.1.1
```

When the `spec.catalog` isn't there it means they have run:
```
$ pctl add --name jack-rocks --repo-url github.com/foo/bar --repo-path foo/ --repo-branch main
$ pctl upgrade jake-rocks v0.1.1
```

if they aren't installing from a catalog, we can't upgrade to `v0.1.1` because we don't have a catalog entry to refer to :smile: 
```

Reply:
ah ok yes that makes sense, thank you for explaining! :)